staging/src/k8s.io/apimachinery/pkg/runtime/codec_test.go

```diff
@@ -39,40 +42,63 @@ func TestCoercingMultiGroupVersioner(t *testing.T) {
 		preferredKinds []schema.GroupKind
 		kinds          []schema.GroupVersionKind
 		expectKind     schema.GroupVersionKind
+		expectedId     string
 	}{
 		{
 			name:           "matched preferred group/kind",
 			target:         gv("mygroup", "__internal"),
 			preferredKinds: []schema.GroupKind{gk("mygroup", "Foo"), gk("anothergroup", "Bar")},
 			kinds:          []schema.GroupVersionKind{gvk("yetanother", "v1", "Baz"), gvk("anothergroup", "v1", "Bar")},
 			expectKind:     gvk("mygroup", "__internal", "Bar"),
+			expectedId:     "{\"accepted\":\"Foo.mygroup,Bar.anothergroup\",\"coerce\":\"true\",\"name\":\"multi\",\"target\":\"mygroup/__internal\"}",
 		},
 		{
 			name:           "matched preferred group",
 			target:         gv("mygroup", "__internal"),
 			preferredKinds: []schema.GroupKind{gk("mygroup", ""), gk("anothergroup", "")},
 			kinds:          []schema.GroupVersionKind{gvk("yetanother", "v1", "Baz"), gvk("anothergroup", "v1", "Bar")},
 			expectKind:     gvk("mygroup", "__internal", "Bar"),
+			expectedId:     "{\"accepted\":\".mygroup,.anothergroup\",\"coerce\":\"true\",\"name\":\"multi\",\"target\":\"mygroup/__internal\"}",
 		},
 		{
 			name:           "no preferred group/kind match, uses first kind in list",
 			target:         gv("mygroup", "__internal"),
 			preferredKinds: []schema.GroupKind{gk("mygroup", ""), gk("anothergroup", "")},
 			kinds:          []schema.GroupVersionKind{gvk("yetanother", "v1", "Baz"), gvk("yetanother", "v1", "Bar")},
 			expectKind:     gvk("mygroup", "__internal", "Baz"),
+			expectedId:     "{\"accepted\":\".mygroup,.anothergroup\",\"coerce\":\"true\",\"name\":\"multi\",\"target\":\"mygroup/__internal\"}",
 		},
 	}
 
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
-			v := NewCoercingMultiGroupVersioner(tc.target, tc.preferredKinds...)
+			v := runtime.NewCoercingMultiGroupVersioner(tc.target, tc.preferredKinds...)
 			kind, ok := v.KindForGroupVersionKinds(tc.kinds)
 			if !ok {
 				t.Error("got no kind")
 			}
 			if kind != tc.expectKind {
 				t.Errorf("expected %#v, got %#v", tc.expectKind, kind)
 			}
+			if e, a := tc.expectedId, v.Identifier(); e != a {
+				t.Errorf("unexpected identifier: %s, expected: %s", a, e)
+			}
 		})
 	}
 }
+
+type mockEncoder struct{}
+
+func (m *mockEncoder) Encode(obj runtime.Object, w io.Writer) error {
+	_, err := w.Write([]byte("mock-result"))
+	return err
+}
+
+func (m *mockEncoder) Identifier() runtime.Identifier {
+	return runtime.Identifier("mock-identifier")
+}
+
+func TestCacheableObject(t *testing.T) {
```

Comment:
if we move this test to runtimetesting (and name it TestCacheableBase64Serializer or something) we don't have to rename this package to break a cycle

Reply:
I was thinking about it a bit more, and I'm not convinced it would actually be better:
- we would be testing bas64 serializer in a different place
- pkg/runtime/testing is the testing infrastructure - putting tests to it will be misleading

[We may create yet another directory for it e.g. pkg/runtime/tests, but it would be an overkill imho]